### ISSUE TITLE
Runtime error when trying to generate letsencrypt certificate with ovh provider

### ISSUE THREAD

**Issue by damiens-robert** [containous/traefik#2736]
### Do you want to request a *feature* or report a *bug*?

I want to report a bug which occurs when I'm trying to configure traefik to use the ovh provider when generating the letsencrypt certificate.

### What did you do?

I configured traefik to use letsencrypt and docker and tried to start it.

### What did you expect to see?

I expected traefik to start and proceed normally

### What did you see instead?

A go runtime error.

### Output of `traefik version`: (_What version of Traefik are you using?_)

I'm using the traefik container image : traefik:v1.5-alpine

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

toml : 

```toml
debug = true

defaultEntryPoints = ["http", "https"]

[web]
address = ":8080"
  [web.auth.basic]
  users = ["admin:$apr1$/cMHxQrz$NrmD8Lhx1dfPGCRjjPxva/"]
[entryPoints]
  [entryPoints.http]
    address = ":80"
    [entryPoints.http.redirect]
      entryPoints = "https"
  [entryPoints.https]
    address = ":443"
    [entryPoints.https.tls]
#      [[entryPoints.https.tls.certificates]]
#      CertFile = "/certs/live/demo1.idemacloud.com/fullchain.pem"
#      KeyFile = "/certs/live/demo1.idemacloud.com/privkey.pem"

[docker]
endpoint= "unix:///var/run/docker.sock"
domain = "idemacloud.com"
watch = true

[acme]
email = "damiens.robert@gmail.com"
storage = "acme.json"
entryPoint = "https"
OnHostRule = true
dnsProvider = "ovh"
```

docker-compose.yml :
```yml

version: '2'

volumes:
        database_data02b:
                driver: local

services:
        php:
                container_name: dockerapache_php_preprod
                restart: always
                image: damiens/httpd:version1
                expose:
                        - "80"
                volumes:
                        - "./www:/usr/local/apache2/www"
                        - "./conf.d/vhost.conf:/usr/local/apache2/conf/extra/httpd-vhosts.conf"
                networks:
                        - traefik_proxy_network
                        - internal
                labels:
                        - traefik.backend=preprod
                        - traefik.frontend.rule=Host:demo2.idemacloud.com,demo3.idemacloud.com
                        - traefik.port=80
                        - traefik.docker.network=traefik_proxy_network
                        - traefik.domain = idemacloud.com
networks:
        traefik_proxy_network:
                external: true
        internal:
                external: false
```

### If applicable, please paste the log output in debug mode (`--debug` switch)

```shell
 time="2018-01-22T20:09:27Z" level=debug msg="LoadCertificateForDomains [demo1.idemacloud.com]..." 
 time="2018-01-22T20:09:27Z" level=debug msg="Looking for provided certificate to validate [demo1.idemacloud.com]..." 
 time="2018-01-22T20:09:27Z" level=debug msg="No provided certificate found for domains [demo1.idemacloud.com], get ACME certificate." 
 time="2018-01-22T20:09:27Z" level=debug msg="Loading ACME certificates [demo1.idemacloud.com]..." 
 time="2018-01-22T20:09:27Z" level=error msg="Error in Go routine: runtime error: invalid memory address or nil pointer dereference" 
 goroutine 26 [running]:
 runtime/debug.Stack(0x74ebff, 0xc4200622d0, 0x2126dee)
 	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
 runtime/debug.PrintStack()
 	/usr/local/go/src/runtime/debug/stack.go:16 +0x22
 github.com/containous/traefik/safe.defaultRecoverGoroutine(0x1dd6920, 0x32e2b30)
 	/go/src/github.com/containous/traefik/safe/routine.go:148 +0x7d
 github.com/containous/traefik/safe.GoWithRecover.func1.1(0x21a9388)
 	/go/src/github.com/containous/traefik/safe/routine.go:139 +0x57
 panic(0x1dd6920, 0x32e2b30)
 	/usr/local/go/src/runtime/panic.go:491 +0x283
 github.com/containous/traefik/vendor/github.com/ovh/go-ovh/ovh.(*Client).CallAPI(0x0, 0x21098a8, 0x4, 0xc420094a50, 0x22, 0x1f56760, 0xc420a32480, 0x1c08fe0, 0xc4208dceb0, 0xc4208fee01, ...)
 	/go/src/github.com/containous/traefik/vendor/github.com/ovh/go-ovh/ovh/ovh.go:283 +0xc7
 github.com/containous/traefik/vendor/github.com/ovh/go-ovh/ovh.(*Client).Post(0x0, 0xc420094a50, 0x22, 0x1f56760, 0xc420a32480, 0x1c08fe0, 0xc4208dceb0, 0x0, 0x0)
 	/go/src/github.com/containous/traefik/vendor/github.com/ovh/go-ovh/ovh/ovh.go:145 +0x94
 github.com/containous/traefik/vendor/github.com/xenolf/lego/providers/dns/ovh.(*DNSProvider).Present(0xc420704ae0, 0xc42069caa5, 0x14, 0xc420587ef0, 0x2b, 0xc420809c20, 0x57, 0x0, 0xc42036a628)
 	/go/src/github.com/containous/traefik/vendor/github.com/xenolf/lego/providers/dns/ovh/ovh.go:97 +0x4bd
 github.com/containous/traefik/vendor/github.com/xenolf/lego/acme.(*dnsChallenge).Solve(0xc420704b00, 0x0, 0x0, 0xc4204d3318, 0x6, 0xc4204d3340, 0x7, 0xc4207b07e0, 0x6a, 0xc420587ef0, ...)
 	/go/src/github.com/containous/traefik/vendor/github.com/xenolf/lego/acme/dns_challenge.go:88 +0x1be
 github.com/containous/traefik/vendor/github.com/xenolf/lego/acme.(*Client).solveChallenges(0xc420330ee0, 0xc420223680, 0x1, 0x1, 0xc420223680)
 	/go/src/github.com/containous/traefik/vendor/github.com/xenolf/lego/acme/client.go:502 +0x123
 github.com/containous/traefik/vendor/github.com/xenolf/lego/acme.(*Client).ObtainCertificate(0xc420330ee0, 0xc4205df8d0, 0x1, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
 	/go/src/github.com/containous/traefik/vendor/github.com/xenolf/lego/acme/client.go:387 +0x305
 github.com/containous/traefik/acme.(*ACME).getDomainsCertificates(0xc42048a5a0, 0xc4205df8d0, 0x1, 0x1, 0x0, 0x0, 0x0)
 	/go/src/github.com/containous/traefik/acme/acme.go:705 +0x1f3
 github.com/containous/traefik/acme.(*ACME).LoadCertificateForDomains.func1()
 	/go/src/github.com/containous/traefik/acme/acme.go:644 +0x4ab
 github.com/containous/traefik/acme.(*ACME).runJobs.func1()
 	/go/src/github.com/containous/traefik/acme/acme.go:724 +0x4e
 github.com/containous/traefik/safe.GoWithRecover.func1(0x21a9388, 0xc4202c8870)
 	/go/src/github.com/containous/traefik/safe/routine.go:142 +0x4d
 created by github.com/containous/traefik/safe.GoWithRecover
 	/go/src/github.com/containous/traefik/safe/routine.go:136 +0x49
```

**Comment by nmengin** [comment]
Hello @damiens-robert.
Many thanks for your interest in the project.

After analyzing deeply the panic error, it seems to be generated when envionment variables are forgotten.

Can you check if the environment variables needed to run a DNSChallenge from OVH are correctly intialized?
All these variables are described [in the documentation](https://docs.traefik.io/configuration/acme/#provider).

Thanks in advance.

**Comment by lubekpl** [comment]
@nmengin I'm gonna use this opportunity to ask ;)

```
traefik_1  | time="2018-02-15T22:09:15Z" level=debug msg="Looking for provided certificate to validate [sub.domain1.eu]..." 
traefik_1  | time="2018-02-15T22:09:15Z" level=debug msg="No provided certificate found for domains [sub.domain1.eu], get ACME certificate." 
traefik_1  | time="2018-02-15T22:09:15Z" level=debug msg="Looking for an existing ACME challenge for sub.domain1.eu..." 
traefik_1  | time="2018-02-15T22:09:15Z" level=debug msg="http: panic serving 89.69.188.237:52158: runtime error: invalid memory address or nil pointer dereference" 
traefik_1  | time="2018-02-15T22:09:15Z" level=debug msg="goroutine 44 [running]:" 
traefik_1  | time="2018-02-15T22:09:15Z" level=debug msg="net/http.(*conn).serve.func1(0xc420a7ae60)" 
traefik_1  | time="2018-02-15T22:09:15Z" level=debug msg="  /usr/local/go/src/net/http/server.go:1697 +0xd0" 
traefik_1  | time="2018-02-15T22:09:15Z" level=debug msg="panic(0x1dd7b00, 0x32e5af0)" 
traefik_1  | time="2018-02-15T22:09:15Z" level=debug msg="  /usr/local/go/src/runtime/panic.go:491 +0x283" 
traefik_1  | time="2018-02-15T22:09:15Z" level=debug msg="github.com/containous/traefik/acme.(*ACME).getCertificate(0xc420528000, 0xc420aba000, 0x1ff4e00, 0xc420aba000, 0x9)" 
traefik_1  | time="2018-02-15T22:09:15Z" level=debug msg="  /go/src/github.com/containous/traefik/acme/acme.go:402 +0x14f" 
traefik_1  | time="2018-02-15T22:09:15Z" level=debug msg="github.com/containous/traefik/acme.(*ACME).(github.com/containous/traefik/acme.getCertificate)-fm(0xc420aba000, 0x32ff8c0, 0x674f43, 0x2109ebe)" 
traefik_1  | time="2018-02-15T22:09:15Z" level=debug msg="  /go/src/github.com/containous/traefik/acme/acme.go:187 +0x34" 
traefik_1  | time="2018-02-15T22:09:15Z" level=debug msg="crypto/tls.(*Config).getCertificate(0xc42097f800, 0xc420aba000, 0x2, 0xc4205be9a0, 0x2)" 
traefik_1  | time="2018-02-15T22:09:15Z" level=debug msg="  /usr/local/go/src/crypto/tls/common.go:717 +0x389" 
traefik_1  | time="2018-02-15T22:09:15Z" level=debug msg="crypto/tls.(*serverHandshakeState).readClientHello(0xc420ac1c08, 0xc420826bf8, 0x5, 0x5)" 
traefik_1  | time="2018-02-15T22:09:15Z" level=debug msg="  /usr/local/go/src/crypto/tls/handshake_server.go:216 +0x3f8" 
traefik_1  | time="2018-02-15T22:09:15Z" level=debug msg="crypto/tls.(*Conn).serverHandshake(0xc420372a80, 0x21b19f8, 0xc420372ba0)" 
traefik_1  | time="2018-02-15T22:09:15Z" level=debug msg="  /usr/local/go/src/crypto/tls/handshake_server.go:48 +0xae" 
traefik_1  | time="2018-02-15T22:09:15Z" level=debug msg="crypto/tls.(*Conn).Handshake(0xc420372a80, 0x0, 0x0)" 
traefik_1  | time="2018-02-15T22:09:15Z" level=debug msg="  /usr/local/go/src/crypto/tls/conn.go:1309 +0x27f" 
traefik_1  | time="2018-02-15T22:09:15Z" level=debug msg="net/http.(*conn).serve(0xc420a7ae60, 0x30ba000, 0xc420ab82d0)" 
traefik_1  | time="2018-02-15T22:09:15Z" level=debug msg="  /usr/local/go/src/net/http/server.go:1713 +0x19a" 
traefik_1  | time="2018-02-15T22:09:15Z" level=debug msg="created by net/http.(*Server).Serve" 
traefik_1  | time="2018-02-15T22:09:15Z" level=debug msg="  /usr/local/go/src/net/http/server.go:2720 +0x288" 
traefik_1  | time="2018-02-15T22:09:16Z" level=error msg="Error getting ACME client: ACME client still not built, retrying in 5.348003787s" 
traefik_1  | time="2018-02-15T22:09:21Z" level=error msg="Error getting ACME client: ACME client still not built, retrying in 6.624381499s" 
traefik_1  | time="2018-02-15T22:09:28Z" level=error msg="Error getting ACME client: ACME client still not built, retrying in 11.602070925s"
```

and my traefik docker-compose looks like this:

```
version: '2'

services:
  traefik:
    restart: always
    image: traefik
    command: --web --docker --docker.domain=local.dev --logLevel=DEBUG
    ports:
      - "80:80"
      - "8080:8080"
      - "443:443"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - ./traefik.toml:/traefik.toml
    networks:
       - traefik_proxy
    environment:
      - OVH_CONSUMER_KEY=my_consumer_key
      - OVH_APPLICATION_KEY=my_app_key
      - OVH_APPLICATION_SECRET=my_app_secret
      - OVH_ENDPOINT=https://eu.api.ovh.com

                              
networks:
  traefik_proxy:
    external: true
```

Is this the right configuration for dnsChallange?

my traefik.toml looks like this:

```
################################################################
# Global configuration
################################################################

# Enable debug mode
#
# Optional
# Default: false
#
# debug = true

# Traefik logs file
# If not defined, logs to stdout
#
# Optional
#
# traefikLogsFile = "log/traefik.log"

# Log level
#
# Optional
# Default: "ERROR"
#
# logLevel = "ERROR"

# Entrypoints to be used by frontends that do not specify any entrypoint.
# Each frontend can specify its own entrypoints.
#
# Optional
# Default: ["http"]
#
defaultEntryPoints = ["http", "https"]

# Entrypoints definition
#
# Optional
# Default:
[entryPoints]
  [entryPoints.http]
  address = ":80"
    [entryPoints.http.redirect]
    entryPoint = "https"
  [entryPoints.https]
  address = ":443"
  [entryPoints.https.tls]

# Enable access logs
# By default it will write to stdout and produce logs in the textual
# Common Log Format (CLF), extended with additional fields.
#
# Optional
#
# [accessLog]

# Sets the file path for the access log. If not specified, stdout will be used.
# Intermediate directories are created if necessary.
#
# Optional
# Default: os.Stdout
#
# filePath = "/path/to/log/log.txt"

# Format is either "json" or "common".
#
# Optional
# Default: "common"
#
# format = "common"

################################################################
# Web configuration backend
################################################################

# Enable web configuration backend
[web]

# Web administration port
#
# Required
#
address = ":8080"

################################################################
# Docker configuration backend
################################################################

# Enable Docker configuration backend
[docker]
#domain = "domain.eu"
watch = true

# Docker server endpoint. Can be a tcp or a unix socket endpoint.
#
# Required
# Default: "unix:///var/run/docker.sock"
#
# endpoint = "tcp://10.10.10.10:2375"

# Default domain used.
# Can be overridden by setting the "traefik.domain" label on a container.
#
# Optional
# Default: ""
#
# domain = "docker.localhost"

# Expose containers by default in traefik
#
# Optional
# Default: true
#
# exposedbydefault = true

[acme]
email = "my@email.info"
storage = "acme.json"
entryPoint = "https"
OnHostRule = true

[acme.dnsChallenge]
  provider = "ovh"
  delayBeforeCheck = 0

```

**Comment by ldez** [comment]
@lubekpl not the good place 😉

Go to the the [Traefik community Slack](https://traefik.herokuapp.com)

**Comment by lubekpl** [comment]
@ldez ok, thanks!

**Comment by devster31** [comment]
I'm also seeing this error with `cloudflare`.
```
traefik    | time="2018-03-17T13:15:06Z" level=info msg="Retrieved ACME certificates"
traefik    | time="2018-03-17T13:15:06Z" level=info msg="Testing certificate renew..."
traefik    | time="2018-03-17T13:15:06Z" level=error msg="Error in Go routine: runtime error: invalid memory address or nil pointer dereference"
traefik    | goroutine 56 [running]:
traefik    | runtime/debug.Stack(0x309204, 0x12d0e360, 0x17cd602)
traefik    | 	/usr/local/go/src/runtime/debug/stack.go:24 +0x80
traefik    | runtime/debug.PrintStack()
traefik    | 	/usr/local/go/src/runtime/debug/stack.go:16 +0x14
traefik    | github.com/containous/traefik/safe.defaultRecoverGoroutine(0x158f0d8, 0x279cf90)
traefik    | 	/go/src/github.com/containous/traefik/safe/routine.go:148 +0x58
traefik    | github.com/containous/traefik/safe.GoWithRecover.func1.1(0x185027c)
traefik    | 	/go/src/github.com/containous/traefik/safe/routine.go:139 +0x44
traefik    | panic(0x158f0d8, 0x279cf90)
traefik    | 	/usr/local/go/src/runtime/panic.go:491 +0x204
traefik    | github.com/containous/traefik/acme.(*DomainsCertificate).needRenew(0x12d9b940, 0x1324f578)
traefik    | 	/go/src/github.com/containous/traefik/acme/account.go:251 +0x18
traefik    | github.com/containous/traefik/acme.(*ACME).renewCertificates.func1()
traefik    | 	/go/src/github.com/containous/traefik/acme/acme.go:462 +0xf4
traefik    | github.com/containous/traefik/acme.(*ACME).runJobs.func1()
traefik    | 	/go/src/github.com/containous/traefik/acme/acme.go:800 +0x38
traefik    | github.com/containous/traefik/safe.GoWithRecover.func1(0x185027c, 0x12d0cb20)
traefik    | 	/go/src/github.com/containous/traefik/safe/routine.go:142 +0x40
traefik    | created by github.com/containous/traefik/safe.GoWithRecover
traefik    | 	/go/src/github.com/containous/traefik/safe/routine.go:136 +0x34
```

My `docker-compose.yml`:
```yaml
services:
  traefik:
    command:
      - --loglevel=INFO
      - --defaultentrypoints=http,https
      - --entryPoints=Name:local Address::800 WhiteListSourceRange:10.0.0.1/16,172.16.0.1/24,::ffff:a00:1/16,::ffff:ac10:1/24
      - --entryPoints=Name:http Address::80 Redirect.EntryPoint:https
      - --entryPoints=Name:https Address::443 Compress:true
          TLS:/certs/${DOMAIN}.crt,/certs/${DOMAIN}.key TLS.MinVersion:VersionTLS12
          TLS.CipherSuites:["TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256","TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384","TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305","TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256","TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384","TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256","TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256"]
      - --entryPoints=Name:api Address::8080
      - --api.entrypoint=api
      - --acme=true
      - --acme.acmelogging=true
      - --acme.email=${EMAIL}
      - --acme.domains=${DOMAIN},home.${DOMAIN}
      - --acme.storage=/etc/traefik/acme.json
      - --acme.entrypoint=https
      - --acme.onhostrule=true
      - --acme.dnschallenge=true
      - --acme.dnschallenge.provider=cloudflare
      - --acme.tlsconfig=true
      - --docker=true
      - --docker.debugloggeneratedtemplate=true
      - --docker.domain='${DOMAIN}'
      - --docker.exposedbydefault=false
      - --docker.usebindportip=true
      - --docker.watch=true
      - --file=true
      - --file.filename=/etc/traefik/external.toml
      - --file.watch=true
    container_name: traefik
    environment:
      CLOUDFLARE_API_KEY: "${CLOUDFLARE_API_KEY}"
      CLOUDFLARE_EMAIL: "${EMAIL}"
    image: traefik
    ports:
      - "80:80"
      - "8080:8080"
      - "443:443"
    restart: unless-stopped
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock"
      - "${MOUNT}/traefik:/etc/traefik"
```

**Comment by dtomcej** [comment]
Hello @damiens-robert,

We have updated the ACME client in our 1.6 release to support the new wildcard functionality from LE.

Is this still an issue? If so, can you retry with the 1.6 release?

Just want to see if we can close this ticket!

Thanks!

**Comment by husimo** [comment]
Hello,
Same error here trying to use OVH api to generate wildcard DNS : 

```toml
defaultEntryPoints = ["http", "https"]
InsecureSkipVerify = true

[entryPoints]
  [entryPoints.http]
  address = ":80"
  [entryPoints.https]
    address = ":443"
    [entryPoints.https.tls]

[acme]
email = "xxx"
storage = "acme.json"
entryPoint = "https"

[acme.dnsChallenge]
  provider = "ovh"
  delayBeforeCheck = 0

[[acme.domains]]
  main = "*.tests.xxx.com"

[web]
address = ":8080"

[docker]
endpoint = "unix:///var/run/docker.sock"
domain = "docker.localhost"
watch = true
exposedbydefault = false
```

```yaml
version: '3'
services:
  traefik-proxy:
    container_name: traefik-proxy
    image: traefik
    restart: always
    networks:
      - traefik-proxy
    ports:
      - "80:80"
      - "443:443"
      - "8080:8080"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - ./traefik.toml:/traefik.toml
      - ./acme.json:/acme.json
    environment:
      - OVH_ENDPOINT='ovh-eu'
      - OVH_APPLICATION_KEY='xxx'
      - OVH_APPLICATION_SECRET='xxx'
      - OVH_CONSUMER_KEY='xxx'
networks:
    traefik-proxy:
      external: true
```

```
traefik-proxy    | time="2018-08-22T15:54:34Z" level=error msg="Error in Go routine: runtime error: invalid memory address or nil pointer dereference"
traefik-proxy    | goroutine 38 [running]:
traefik-proxy    | runtime/debug.Stack(0x737c5f, 0xc4200a62d0, 0x21e7127)
traefik-proxy    |      /usr/local/go/src/runtime/debug/stack.go:24 +0xa7
traefik-proxy    | runtime/debug.PrintStack()
traefik-proxy    |      /usr/local/go/src/runtime/debug/stack.go:16 +0x22
traefik-proxy    | github.com/containous/traefik/safe.defaultRecoverGoroutine(0x1df1bc0, 0x35be280)
traefik-proxy    |      /go/src/github.com/containous/traefik/safe/routine.go:148 +0x73
traefik-proxy    | github.com/containous/traefik/safe.GoWithRecover.func1.1(0x2286870)
traefik-proxy    |      /go/src/github.com/containous/traefik/safe/routine.go:139 +0x57
traefik-proxy    | panic(0x1df1bc0, 0x35be280)
traefik-proxy    |      /usr/local/go/src/runtime/panic.go:502 +0x229
traefik-proxy    | github.com/containous/traefik/vendor/github.com/ovh/go-ovh/ovh.(*Client).NewRequest(0x0, 0x21c5b33, 0x4, 0xc42092c570, 0x21, 0x1fcd860, 0xc420954a80, 0x1, 0xc420954a80, 0x0, ...)
traefik-proxy    |      /go/src/github.com/containous/traefik/vendor/github.com/ovh/go-ovh/ovh/ovh.go:237 +0x97
traefik-proxy    | github.com/containous/traefik/vendor/github.com/ovh/go-ovh/ovh.(*Client).CallAPI(0x0, 0x21c5b33, 0x4, 0xc42092c570, 0x21, 0x1fcd860, 0xc420954a80, 0x1bcb700, 0xc4208aa410, 0xc420a07001, ...)
traefik-proxy    |      /go/src/github.com/containous/traefik/vendor/github.com/ovh/go-ovh/ovh/ovh.go:306 +0x83
traefik-proxy    | github.com/containous/traefik/vendor/github.com/ovh/go-ovh/ovh.(*Client).Post(0x0, 0xc42092c570, 0x21, 0x1fcd860, 0xc420954a80, 0x1bcb700, 0xc4208aa410, 0x0, 0x0)
traefik-proxy    |      /go/src/github.com/containous/traefik/vendor/github.com/ovh/go-ovh/ovh/ovh.go:147 +0x94
traefik-proxy    | github.com/containous/traefik/vendor/github.com/xenolf/lego/providers/dns/ovh.(*DNSProvider).Present(0xc4209ef9c0, 0xc420710420, 0x16, 0xc42092c420, 0x2b, 0xc42009e9c0, 0x57, 0x0, 0x2e)
traefik-proxy    |      /go/src/github.com/containous/traefik/vendor/github.com/xenolf/lego/providers/dns/ovh/ovh.go:97 +0x492
traefik-proxy    | github.com/containous/traefik/vendor/github.com/xenolf/lego/acme.(*dnsChallenge).Solve(0xc4209ef9e0, 0xc4201aa700, 0x6a, 0xc42092a8a0, 0x6, 0xc42092a8a6, 0x7, 0xc42092c420, 0x2b, 0x0, ...)
traefik-proxy    |      /go/src/github.com/containous/traefik/vendor/github.com/xenolf/lego/acme/dns_challenge.go:87 +0x1a1
traefik-proxy    | github.com/containous/traefik/vendor/github.com/xenolf/lego/acme.(*Client).solveChallengeForAuthz(0xc420158540, 0xc420baaae0, 0x1, 0x1, 0x1, 0x1)
traefik-proxy    |      /go/src/github.com/containous/traefik/vendor/github.com/xenolf/lego/acme/client.go:543 +0x1fa
traefik-proxy    | github.com/containous/traefik/vendor/github.com/xenolf/lego/acme.(*Client).ObtainCertificate(0xc420158540, 0xc4209ebde0, 0x1, 0x1, 0x1, 0x1f02c80, 0xc42095e000, 0x0, 0x7f36771156c8, 0x0, ...)
traefik-proxy    |      /go/src/github.com/containous/traefik/vendor/github.com/xenolf/lego/acme/client.go:398 +0x2a5
traefik-proxy    | github.com/containous/traefik/vendor/github.com/xenolf/lego/acme.(*Client).RenewCertificate(0xc420158540, 0xc42036d9a0, 0x16, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc4207fad80, ...)
traefik-proxy    |      /go/src/github.com/containous/traefik/vendor/github.com/xenolf/lego/acme/client.go:499 +0x5bf
traefik-proxy    | github.com/containous/traefik/provider/acme.(*Provider).renewCertificates(0x35da640)
traefik-proxy    |      /go/src/github.com/containous/traefik/provider/acme/provider.go:481 +0x35f
traefik-proxy    | github.com/containous/traefik/provider/acme.(*Provider).Provide(0x35da640, 0xc420095500, 0xc4206dc000, 0x35fba58, 0x0, 0x0, 0xc4205de778, 0x1a5ed24)
traefik-proxy    |      /go/src/github.com/containous/traefik/provider/acme/provider.go:387 +0x1cd
traefik-proxy    | github.com/containous/traefik/configuration.providerAggregator.Provide.func1()
traefik-proxy    |      /go/src/github.com/containous/traefik/configuration/provider_aggregator.go:90 +0x72
traefik-proxy    | github.com/containous/traefik/safe.GoWithRecover.func1(0x2286870, 0xc4205394a0)
traefik-proxy    |      /go/src/github.com/containous/traefik/safe/routine.go:142 +0x4d
traefik-proxy    | created by github.com/containous/traefik/safe.GoWithRecover
traefik-proxy    |      /go/src/github.com/containous/traefik/safe/routine.go:136 +0x49
traefik-proxy    | time="2018-08-22T15:54:34Z" level=error msg="Error in Go routine: runtime error: invalid memory address or nil pointer dereference"
traefik-proxy    | goroutine 41 [running]:
traefik-proxy    | runtime/debug.Stack(0x737c5f, 0xc4200a62d0, 0x21e7127)
traefik-proxy    |      /usr/local/go/src/runtime/debug/stack.go:24 +0xa7
traefik-proxy    | runtime/debug.PrintStack()
traefik-proxy    |      /usr/local/go/src/runtime/debug/stack.go:16 +0x22
traefik-proxy    | github.com/containous/traefik/safe.defaultRecoverGoroutine(0x1df1bc0, 0x35be280)
traefik-proxy    |      /go/src/github.com/containous/traefik/safe/routine.go:148 +0x73
traefik-proxy    | github.com/containous/traefik/safe.GoWithRecover.func1.1(0x2286870)
traefik-proxy    |      /go/src/github.com/containous/traefik/safe/routine.go:139 +0x57
traefik-proxy    | panic(0x1df1bc0, 0x35be280)
traefik-proxy    |      /usr/local/go/src/runtime/panic.go:502 +0x229
traefik-proxy    | github.com/containous/traefik/vendor/github.com/ovh/go-ovh/ovh.(*Client).NewRequest(0x0, 0x21c5b33, 0x4, 0xc42092c720, 0x21, 0x1fcd860, 0xc420954e80, 0x1, 0xc420954e80, 0x0, ...)
traefik-proxy    |      /go/src/github.com/containous/traefik/vendor/github.com/ovh/go-ovh/ovh/ovh.go:237 +0x97
traefik-proxy    | github.com/containous/traefik/vendor/github.com/ovh/go-ovh/ovh.(*Client).CallAPI(0x0, 0x21c5b33, 0x4, 0xc42092c720, 0x21, 0x1fcd860, 0xc420954e80, 0x1bcb700, 0xc4208aa5f0, 0xc420c11401, ...)
traefik-proxy    |      /go/src/github.com/containous/traefik/vendor/github.com/ovh/go-ovh/ovh/ovh.go:306 +0x83
traefik-proxy    | github.com/containous/traefik/vendor/github.com/ovh/go-ovh/ovh.(*Client).Post(0x0, 0xc42092c720, 0x21, 0x1fcd860, 0xc420954e80, 0x1bcb700, 0xc4208aa5f0, 0x0, 0x0)
traefik-proxy    |      /go/src/github.com/containous/traefik/vendor/github.com/ovh/go-ovh/ovh/ovh.go:147 +0x94
traefik-proxy    | github.com/containous/traefik/vendor/github.com/xenolf/lego/providers/dns/ovh.(*DNSProvider).Present(0xc4209ef9c0, 0xc4207106e0, 0x13, 0xc42092c5d0, 0x2b, 0xc42009ec60, 0x57, 0x0, 0xc420c115b0)
traefik-proxy    |      /go/src/github.com/containous/traefik/vendor/github.com/xenolf/lego/providers/dns/ovh/ovh.go:97 +0x492
traefik-proxy    | github.com/containous/traefik/vendor/github.com/xenolf/lego/acme.(*dnsChallenge).Solve(0xc4209ef9e0, 0xc4201aa8c0, 0x6a, 0xc42092aaca, 0x6, 0xc42092aad0, 0x7, 0xc42092c5d0, 0x2b, 0x0, ...)
traefik-proxy    |      /go/src/github.com/containous/traefik/vendor/github.com/xenolf/lego/acme/dns_challenge.go:87 +0x1a1
traefik-proxy    | github.com/containous/traefik/vendor/github.com/xenolf/lego/acme.(*Client).solveChallengeForAuthz(0xc420158540, 0xc420baacc0, 0x1, 0x1, 0x1, 0x1)
traefik-proxy    |      /go/src/github.com/containous/traefik/vendor/github.com/xenolf/lego/acme/client.go:543 +0x1fa
traefik-proxy    | github.com/containous/traefik/vendor/github.com/xenolf/lego/acme.(*Client).ObtainCertificate(0xc420158540, 0xc420605280, 0x1, 0x1, 0x1, 0x0, 0x0, 0x0, 0x1, 0x1, ...)
traefik-proxy    |      /go/src/github.com/containous/traefik/vendor/github.com/xenolf/lego/acme/client.go:398 +0x2a5
traefik-proxy    | github.com/containous/traefik/provider/acme.(*Provider).resolveCertificate(0x35da640, 0xc420541720, 0x15, 0xc4205cde40, 0x0, 0x0, 0x1, 0x0, 0x0, 0x0)
traefik-proxy    |      /go/src/github.com/containous/traefik/provider/acme/provider.go:245 +0x32c
traefik-proxy    | github.com/containous/traefik/provider/acme.(*Provider).Provide.func1()
traefik-proxy    |      /go/src/github.com/containous/traefik/provider/acme/provider.go:381 +0x7c
traefik-proxy    | github.com/containous/traefik/safe.GoWithRecover.func1(0x2286870, 0xc420611260)
traefik-proxy    |      /go/src/github.com/containous/traefik/safe/routine.go:142 +0x4d
traefik-proxy    | created by github.com/containous/traefik/safe.GoWithRecover
traefik-proxy    |      /go/src/github.com/containous/traefik/safe/routine.go:136 +0x49
```

Thank you :-)

**Comment by jseguillon** [comment]
Had the same problem while testing. Explanation seems quite simple : if you test with incorrectly formated values like "foo" or "bar" => you'll get this error. 

If you use real values with good length/format there will be no problem (hum at least that was MY problem). 

So issue seems more something like "Traefik should test keys format and warn instead of crashing".

**Comment by juxorevo** [comment]
Hi husimo, 

I solve my problem by adding manualy the TXT in DNS.

My configuration now : 

version: '3'
services:
  reverseproxy:
    image: arm32v6/traefik:maroilles
    command: --web --docker --docker.domain=$(domain) --docker.watch \
      --defaultEntryPoints='http,https' \
      --entryPoints='Name:http Address::80 Redirect.EntryPoint:https' \
      --entryPoints='Name:https Address::443 TLS' \
        entrypoint
      --docker.swarmmode=true \
      --docker.exposedbydefault=false \
      --acme \
      --acme.entryPoint='https' \
      --acme.email='be.aimard@gmail.com' \
      --acme.ondemand=false \
      --acme.acmelogging=true \
      --acme.onhostrule=true \
      --acme.storage='/etc/traefik/acme/acme.json'
      --acme.dnsChallenge.provider='manual'
      --acme.domains='*.$(domain)'
    networks:
      - web
    ports:
      - "80:80"
      - "8091:8080"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - /mnt/rasp/traefik/acme:/etc/traefik/acme:rw
      - /dev/null:/traefik.toml
    deploy:
      placement:
        constraints:
         - node.role == manager

    #      --logLevel=DEBUG \
networks:
  web:
    external: true
# Redirect.EntryPoint:https' \

**Comment by husimo** [comment]
@juxorevo How do I know what i'm supposed to put in TXT ?Thanks

**Comment by traefiker** [comment]
Closed by #3888.
